roles/StackStorm.mongodb/tasks/main.yml

```diff
@@ -1,9 +1,20 @@
 ---
-- name: Retrieve variables
+- name: Retrieve variables (RedHat)
   include_vars: "{{ item }}"
   with_first_found:
     - "{{ ansible_os_family | lower }}_{{ ansible_distribution_major_version }}.yml"
     - "{{ ansible_os_family | lower }}.yml"
+  when: ansible_facts.os_family == 'RedHat'
+  tags: [databases, mongodb]
+
+- name: Retrieve variables (Debian)
+  include_vars: debian.yml
+  when: ansible_facts.os_family == 'Debian'
+  tags: [databases, mongodb]
+
+- name: Retrieve variables (Debian)
+  include_vars: debian_20.yml
+  when: ansible_facts.os_family == 'Debian' and ansible_facts.distribution_major_version == '20'
   tags: [databases, mongodb]
```

Comment:
This 3-tasks construct doesn't look good indeed.
Avoiding os-specific `when` condition is exactly the purpose of the 1st task `with_first_found` which is a known best practice.

Reply:
That's true and imho it's actually an anti-pattern to override defaults via role variables. This has a high potential to cause failures and confuse maintainers. Now I switched to the solution suggested by @cognifloyd which seems to be the most elegant one.